splinterd/src/daemon.rs

```diff
@@ -982,6 +984,35 @@ fn start_health_service(
 fn create_store_factory(
     db_url: &str,
 ) -> Result<Box<dyn splinter::store::StoreFactory>, StartError> {
+    // If the given db_url is a file path check that the file exists and is not empty
+    if !db_url.starts_with("postgres://") & (db_url != "memory") & (db_url != ":memory:") {
+        if !Path::new(db_url).exists() {
+            return Err(StartError::StorageError(format!(
+                "Database file '{}' does not exist",
+                db_url
+            )));
+        } else {
+            let mut file = File::open(db_url).map_err(|err| {
+                StartError::StorageError(format!(
+                    "Unable to open database file '{}' : {}",
+                    db_url, err
+                ))
+            })?;
+            let mut buffer = Vec::new();
+            file.read_to_end(&mut buffer).map_err(|err| {
```

Comment:
This will not work for very large databases as it will be constrained by the size of memory.

It is probably sufficient here to check for the existence of the file (Path::new().exists()) in order to ensure we are not going to accidentally create an empty database.

Reply:
Removed this from the check so now only the existence of the file is checked